### PR TITLE
refactor: repository-list-item to styled components

### PR DESCRIPTION
**Pull request by apoeco**
Related #532 

## Screenshots
| Before   | After    |
| -------- | -------- |
| before | after |

### CODE REVIEW

**Comment by andrewda** [comment]
Could you add before/after screenshots, as provided in the PR template?

File: src/components/repository-list-item.component.js [hunk 1]
```diff
@@ -1,119 +1,127 @@
...
+  </ListItemWrapper>
 );
 
+const styles = StyleSheet.create({
+  title: {
+    color: colors.primaryDark,
+    fontFamily: styledFonts.fontPrimarySemiBold,
+  },
+});
+
```
Comment (machour) [inline src/components/repository-list-item.component.js]: We should be able to get rid of this last one when #619 gets merged.

The `<ListItem>` can be then styled as follow:

```js
const Repository = styled(ListItem).attrs({
  titleStyle: {
    color: colors.primaryDark,
    fontFamily: styledFonts.fontPrimarySemiBold,
  },
})``;
```

the `underlayColor` prop can also be merged there as it's is a pure presentation prop

Comment (machour) [reply src/components/repository-list-item.component.js]: #619 is merged, you should now be able to get rid of this last thingy 🎉

**Comment by apoeco** [comment]
I think everything is ready now 😃

**Comment by machour** [comment]
Hi @apoeco and sorry for not getting back to you sooner 😞 

Just had time to review this, and it's working flawlessly 🎉 

I wanted to fix the conflicts for you, but it seems that I'm unable to do it because you didn't work on a branch of your fork, but rather on master directly.  Could you take care of fixing them ? (You just need to retrieve the last `.all-contributorsrc` version and add yourself again, both .md files will be regenerated)

We'll get this merged as soon as it's done, thank you again!

File: src/components/repository-list-item.component.js [hunk 2]
```diff
@@ -1,139 +1,144 @@
...
+  flex-direction: row;
+  align-items: center;
+`;
+
+const ColoredText = styled.Text`
+  color: ${colors.primaryDark};
+`;
+
+const TitleText = ColoredText.extend`
+  font-family: ${styledFonts.fontPrimarySemiBold};
```
Comment (machour) [inline src/components/repository-list-item.component.js]: Should be:

```
const TitleText = ColoredText.extend`
  ${fonts.fontPrimarySemiBold};
`;
```

File: src/components/repository-list-item.component.js [hunk 3]
```diff
@@ -1,139 +1,144 @@
...
+`;
+
+const PrivateIconContainer = styled.View`
+  margin-left: 6;
+`;
+
+const DescriptionText = ColoredText.extend`
+  font-family: ${styledFonts.fontPrimaryLight};
```
Comment (machour) [inline src/components/repository-list-item.component.js]: Should be:

```
const DescriptionText = ColoredText.extend`
  ${fonts.fontPrimaryLight};
`;
```

File: src/components/repository-list-item.component.js [hunk 4]
```diff
@@ -1,139 +1,144 @@
...
+  flex: 1;
+  padding-top: 5;
+`;
+
+const ExtraInfoText = styled.Text`
+  color: ${colors.greyDark};
+  padding-left: 3;
+  padding-top: 2;
+  margin-right: 15;
+  font-family: ${styledFonts.fontPrimary};
```
Comment (machour) [inline src/components/repository-list-item.component.js]: ${fonts.fontPrimary};

File: src/components/repository-list-item.component.js [hunk 5]
```diff
@@ -1,139 +1,144 @@
...
-  </View>
+      <ExtraInfoText>{repository.language}</ExtraInfoText>
+    </ExtraInfoWrapper>
+  </ListItemWrapper>
 );
 
+const Repository = styled(ListItem).attrs({
+  titleStyle: {
+    color: colors.primaryDark,
+    fontFamily: styledFonts.fontPrimarySemiBold,
```
Comment (machour) [inline src/components/repository-list-item.component.js]: fontFamily: fonts.fontPrimarySemiBold.fontFamily,

File: src/components/repository-list-item.component.js [hunk 6]
```diff
@@ -1,139 +1,144 @@
 import React from 'react';
-import { StyleSheet, Text, View } from 'react-native';
+import styled from 'styled-components/native';
 import { ListItem, Icon } from 'react-native-elements';
 
 import { emojifyText, abbreviateNumber } from 'utils';
-import { colors, languageColors, fonts, normalize } from 'config';
+import { colors, languageColors, styledFonts, normalize } from 'config';
```
Comment (machour) [inline src/components/repository-list-item.component.js]: use `fonts` instead of `styledFonts`

**Comment by apoeco** [comment]
I’m on it!

File: src/components/repository-list-item.component.js [hunk 7]
```diff
@@ -54,7 +54,7 @@ const ExtraInfoText = styled.Text`
   padding-left: 3;
   padding-top: 2;
   margin-right: 15;
-  font-family: ${styledFonts.fontPrimary};
+  font-family: ${fonts.fontPrimary};
```
Comment (machour) [inline src/components/repository-list-item.component.js]: Almost :D 
Remove `font-family: ` and we're good to go

**Comment by machour** [comment]
Thank you!!! 🎉 🎉 🎉 🎉 🎉

**Pull request merged**